swap/src/protocol/alice/swap.rs

```diff
@@ -85,11 +85,14 @@ async fn run_until_internal(
                 .await
                 .context("Failed to find lock Bitcoin tx")??;
 
-                bitcoin_wallet
-                    .watch_until_status(&state3.tx_lock, |status| {
+                timeout(
+                    env_config.bitcoin_lock_confirmed_timeout(),
+                    bitcoin_wallet.watch_until_status(&state3.tx_lock, |status| {
                         status.is_confirmed_with(env_config.bitcoin_finality_confirmations)
-                    })
-                    .await?;
+                    }),
+                )
+                .await
+                .context("Lock Bitcoin TX failed to confirm in time")??;
```

Comment:
Isn't that exactly what happens a couple of lines above?

We first check that we see the TX in mempool within the given timeout and then we wait for confirmations.

Reply:
Do you mean the .context message? I just pushed an amended commit to improve it

Reply:
No, I mean the whole `timeout` call. We literally have the same timeout a couple of lines above where we wait for the transaction to be seen.

I also think that this is actually wrong. It is fine to race a timeout against _seeing_ the transaction but waiting for finality can take quite a bit longer than the now configured 10 minutes. What if we configure finality for more than one block, which is what would happen on mainnet?

Reply:
Ah crap I was supposed to use this function
```
pub fn bitcoin_lock_confirmed_timeout(&self) -> Duration {
         self.bitcoin_avg_block_time * self.bitcoin_finality_confirmations * 4
}
```

Reply:
Ah I didn't see that these are actually two different timeouts. My bad.

I am not fully convinced to be honest, do we really need to timeouts here or should we just wait for the transaction to be confirmed? I guess it might fall out of the mempool or something but that is pretty unlikely.

I'd rather keep it simple for now to be honest.

Reply:
If we don't timeout that means we have a "state" that we can get stuck in. Seems like a bad design to me. Maybe a better approach would be to timeout if the transaction falls out of the mempool: goes to unseen after being seen in the mempool?